MediaGallerySynchronization/Model/SelectByBatchesGenerator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\MediaGallerySynchronization\Model;
+
+use Magento\Framework\App\ResourceConnection;
+use Psr\Log\LoggerInterface;
+use Magento\MediaGallerySynchronizationApi\Model\FetchBatchesInterface;
+
+/**
+ * Select data from database by provided batch size
+ */
+class SelectByBatchesGenerator implements FetchBatchesInterface
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var int
+     */
+    private $batchSize;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+    
+    /**
+     * @param ResourceConnection $resourceConnection
+     * @param LoggerInterface $logger
+     * @param int $batchSize
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        LoggerInterface $logger,
+        int $batchSize
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->logger = $logger;
+        $this->batchSize = $batchSize;
+    }
+
+    /**
+     * Get data from table by batches, based on limit offset value.
+     *
+     * @param string $tableName
+     * @param array $columns
+     */
+    public function execute(string $tableName, array $columns): \Traversable
+    {
+        try {
+            $connection = $this->resourceConnection->getConnection();
+            $tableName = $this->resourceConnection->getTableName($tableName);
+            $total =  $connection->fetchOne($connection->select()->from($tableName, 'COUNT(*)'));
+            $batches = ceil($total / $this->batchSize);
+
+            for ($i = 0; $i < $batches; $i++) {
```

Comment:
What do you think about giving the `$i` and `$batches` a more meaningful name
```suggestion
            for ($page = 0; $page < $totalPages; $page++) {
```
(or `$batchNumber`, `$totalBatches`)
I think even the variables naming seems to be important in this implementation for better readability and this is not common implementation for Magento codebase